subprojects/packagefiles/sdl2/meson.build

```diff
@@ -367,11 +368,21 @@ endif
 core_deps = [
     cc.find_library('m', required : false),
     cc.find_library('dl', required : false),
-    cc.find_library('iconv', required : false),
     dbus_dep,
     ibus_dep,
 ]
 
+if not opt_system_iconv.disabled()
+    iconv_dep = cc.find_library('iconv', required : opt_system_iconv)
+else
+    iconv_dep = fake_dep
+endif
+
+if cc.has_function('iconv_open', dependencies: iconv_dep)
```

Comment:
to me this looks sound in theory.
But when testing it said it tested for `iconv_open` while linking with `-liconv`.
I have no libiconv on my system, was this only a visual aspect or did it link regardless of if it found anything?

Reply:
Meson has a built-in `dependency('iconv')` that handles this correctly. But using it requires a fairly recent Meson.

You can emulate it for older versions of Meson by checking for iconv_open without any dependencies, and then if that fails, checking for it with `cc.find_library('iconv')`. Additional handling is needed to manage feature options correctly, and of course some systems do really weird things with has_function returning a dummy built-in that doesn't actually link, so you don't want to use has_function after all -- instead you want `cc.links()` on a crafted code snippet that actually includes iconv.h and runs a no-frills `int main() { iconv_open(...); }`.

It's all kinds of super fun, that's why I added the built-in dependency handling. :P